src/scipp/coords/transform_coords.py

```diff
@@ -75,17 +82,30 @@ def transform_coords(x: Union[DataArray, Dataset],
                       keep_intermediate=keep_intermediate,
                       keep_inputs=keep_inputs,
                       quiet=quiet)
-    targets = {targets} if isinstance(targets, str) else set(targets)
-    if isinstance(x, DataArray):
-        return _transform_data_array(x,
-                                     targets=targets,
-                                     graph=Graph(graph),
-                                     options=options)
+    if targets is None:
+        targets = []
+    if graph is None:
+        graph = {}
+
+    if isinstance(targets, dict):
+        if graph:
+            raise ValueError(
+                "Got multiple transformation graphs, must provide only one.")
+        graph = targets
+        targets = set(targets)
     else:
-        return _transform_dataset(x,
-                                  targets=targets,
-                                  graph=Graph(graph),
-                                  options=options)
+        targets = {targets} if isinstance(targets, str) else set(targets)
+
+    if not set(kwargs).isdisjoint(graph):
+        raise ValueError(f"Duplicate graph nodes in transformation graph {list(graph)} "
+                         f"and keyword arguments {list(kwargs)}.")
```

Comment:
I had briefly considered letting keyword-args replace nodes in the graph (basically what `update` does below, without this check), but I feel that would be a bad idea?